engine/base/nu-kafka-engine-bin-test/src/it/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineBinTest.scala

```diff
@@ -0,0 +1,89 @@
+package pl.touk.nussknacker.engine.baseengine.kafka
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.commons.io.FileUtils
+import org.scalatest.{FunSuite, Matchers}
+import org.springframework.util.StreamUtils
+import pl.touk.nussknacker.engine.build.EspProcessBuilder
+import pl.touk.nussknacker.engine.canonize.ProcessCanonizer
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.kafka.KafkaSpec
+import pl.touk.nussknacker.engine.marshall.ProcessMarshaller
+import pl.touk.nussknacker.engine.spel.Implicits._
+import pl.touk.nussknacker.test.VeryPatientScalaFutures
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+import scala.concurrent.Promise
+import scala.util.Try
+
+class NuKafkaEngineBinTest extends FunSuite with KafkaSpec with Matchers with LazyLogging with VeryPatientScalaFutures {
+
+  private val inputTopic = s"${getClass.getSimpleName}-input"
+
+  private val outputTopic = s"${getClass.getSimpleName}-output"
+
+  // FIXME: fix NoClassDefFoundError: org/apache/flink/streaming/api/functions/sink/SinkFunction
+  ignore("should run scenario and pass data to output ") {
+    kafkaClient.createTopic(inputTopic)
+    kafkaClient.createTopic(outputTopic, 1)
+
+    val scenario = EspProcessBuilder
+        .id("test")
+        .exceptionHandler()
+        .source("source", "kafka-json", "topic" -> s"'$inputTopic'")
+        .emptySink("sink", "kafka-json", "topic" -> s"'$outputTopic'", "value" -> "#input")
+    val jsonFile = saveScenarioToTmp(scenario)
+
+    val engineExitCodePromise = Promise[Int]
```

Comment:
I wonder, will it be possible to use dockerPublishLocal + TestContainers for this test in the future?

Reply:
I would like to do this in separate PR

Reply:
:+1: